ibis/expr/types/relations.py

```diff
@@ -2370,6 +2369,79 @@ def to_pandas(self, **kwargs) -> pd.DataFrame:
         """
         return self.execute(**kwargs)
 
+    def cache(self) -> Table:
+        """Cache the provided expression.
+
+         All subsequent operations on the returned expression will be performed on the cached data.
+         You may use the ``with`` statement to explicitly clean up the cached expression when it's not needed.
+
+        !!! note "This method eagerly evaluates the expression prior to caching"
+
+        Returns
+        -------
+        Table
+            Cached table.
+
+        Examples
+        --------
+        >>> import ibis
+        >>> ibis.options.interactive = True
+        >>> t = ibis.examples.penguins.fetch(table_name="penguins")
+        >>> cached_penguins = t.cache()
+        >>> cached_penguins
```

Comment:
I wonder if turning off interactive mode would offer a more illustrative example, since the repr of the cached table would have the word "cache" in the table name.

As this is written right now, it's more of an example of how to fetch an example dataset.

Reply:
@gforsyth, I can do show_sql() instead and show how the select is from the cached table. 
but it is very specific to a DB based backends.

what I think is better is -  `t.mutate(calculation="very slow calculation ;-)").cache()` to show how potentially complex calculation can be cached.

WDYT?

Reply:
The `mutate` example sounds good to me!